File: leetcode/kotlin/14.LongestCommonPrefix.kt
package com.programming

internal class Test{
    companion object{
        fun lcp(s: List<String>):String {
            if(s.isEmpty()){
                return ""
            }
            var prefix:String = s[0]
            for(i in s.indices){
                for(j in prefix.indices){
                    if(j == s[i].length || prefix[j]!= s[i][j]){
                        prefix = prefix.substring(0,j)
                        break
                    }
                }
            }
            return prefix
        }
    }
}

fun main(){
    print(Test.lcp(listOf<String>("flower", "flow", "flight")))
}

File: leetcode/python/14.LongestCommonPrefix.py
class Solution:

  @staticmethod
  def lcp(s):
    if len(s)==0:
      return ""
    prefix = s[0]
    for i in range(len(s)):
      for j in range(len(prefix)):
        if j == len(s[i]) or prefix[j] != s[i][j]:
          prefix = prefix[:j]
          break
    return prefix


print(Solution.lcp(["flower", "flow", "flight"]))
